### PR TITLE
Add more tests for TestPrestoSparkHttpWorkerClient

### PR DESCRIPTION
**Pull request by tanjialiang**
Adds a couple of more thorough tests to TestPrestoSparkHttpWorkerClient
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 1]
```diff
@@ -101,6 +104,19 @@
     private static final JsonCodec<PlanFragment> PLAN_FRAGMENT_JSON_CODEC = JsonCodec.jsonCodec(PlanFragment.class);
     private static final JsonCodec<TaskUpdateRequest> TASK_UPDATE_REQUEST_JSON_CODEC = JsonCodec.jsonCodec(TaskUpdateRequest.class);
 
+    private static SerializedPage createSerializedPage(int numBytes)
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:107]: move private helpers to the end of the file

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 2]
```diff
@@ -261,13 +277,312 @@ public void testResultFetcher()
     }
 
     @Test
-    public void testResultFetcherFail()
+    public void testResultFetcherMultipleNonEmptyResults()
+    {
+        TaskId taskId = new TaskId("testid", 0, 0, 0);
+        URI uri = uriBuilderFrom(BASE_URI).appendPath(TASK_ROOT_PATH).build();
+        final int serializedPageSize = 1 * 1024 * 1024; // 1MB
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:284]: nit: 
```java
new DataSize(1, MEGABYTE).toBytes()

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 3]
```diff
@@ -261,13 +277,312 @@ public void testResultFetcher()
...
+        {
+            return numPages - requestCount;
+        }
+    }
+
+    @Test
+    public void testResultFetcherExceedingBufferLimit()
+    {
+        final int numPages = 10;
+        final int serializedPageSize = 32 * 1024 * 1024;
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:403]: same

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 4]
```diff
@@ -261,13 +277,312 @@ public void testResultFetcher()
...
+        public int getRemainingPageCount()
+        {
+            return numPages - requestCount;
+        }
+    }
+
+    @Test
+    public void testResultFetcherExceedingBufferLimit()
+    {
+        final int numPages = 10;
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:402]: we don't use `final` for temp variables; remove `final`

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 5]
```diff
@@ -261,13 +277,312 @@ public void testResultFetcher()
...
+            }
+            // Wait a bit for fetches to overwhelm memory.
+            Thread.sleep(5000);
+            assertEquals(breakingLimitResponseManager.getRemainingPageCount(), 5);
+            List<SerializedPage> pages = new ArrayList<>();
+            pages.add(page.get());
+            while (pages.size() < numPages) {
+                page = taskResultFetcher.pollPage();
+                if (page.isPresent()) {
+                    pages.add(page.get());
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:438]: `page.ifPresent(pages::add);`

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 6]
```diff
@@ -628,6 +926,83 @@ public static Response createTaskInfoResponse(HttpStatus httpStatus, String task
...
+                headers.put(HeaderName.of(CONTENT_TYPE), PRESTO_PAGES_TYPE.toString());
+                return new TestingResponse(
+                        httpStatus.code(),
+                        httpStatus.toString(),
+                        headers,
+                        slicedOutput.slice().getInput());
+            }
+        }
+
+        public static interface DelayGenerator
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:973]: remove static

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java [hunk 7]
```diff
@@ -628,6 +926,83 @@ public static Response createTaskInfoResponse(HttpStatus httpStatus, String task
...
+        {
+            public Duration getDelay();
```
Comment (highker) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:975]: remove public

**Review by highker — APPROVED** [review]
please make sure tests are passing before merging

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java (hunk 4 above)
Comment (tanjialiang) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java:402]: Sure. But mind if I ask why?

**Pull request merged**